docs/20_smart-contracts/40_guides/10_create_smart_contracts_guide.md

```diff
@@ -0,0 +1,1351 @@
+
+```cpp
+balance_table balances(get_self(), get_self().value);
+```
+
+In the code above, the first parameter, the`code`, is initialized with the `get_self()`, witch returns the contract account owner. The second parameter, the `scope`, is initialized with the `get_self().value`, which returns the same account value as the contract owner.
+
+Important to note is that the code above creates a `balances` object which is an accessor to the multi-index `balances` table, which is an address of the RAM storage space where the table rows/objects are saved.
+
+### Multi-index: Find in Table
+
+This is how to query the `balance_table` based on its primary key:
+
+```cpp
+balance_table balances(get_self(), get_self().value);
+
+auto itr = balances.find(name.value);
+if ( itr != balances.end() ) {
+    // entity  found by name
+}
+else {
+    // entity not found
+}
+```
+
+### Multi-index: Insert in Table
+
+This is how to insert a row into the `balance_table`:
+
+```cpp
+balance_table balances(get_self(), get_self().value);
+
+balances.emplace(get_self(), [&](auto& row) {
+  row.user = "user123"_n;
+  row.balance = asset(1, symbol("EOS", 4));
+});
+```
+
+The code above uses emplace method to insert a new row into the table, and specifies the user's account name and balance as the values for the row's fields.
+
+### Multi-index: Modify Existing Data
+
+This is how to modify an existing row in the `balance_table`:
+
+```cpp
+balance_table balances(get_self(), get_self().value);
+
+auto itr = balances.find(name.value);
+if ( itr != balances.end() ) {
+    balances.modify(itr, get_self(), [&](auto& row) {
+        row.balance = asset(1, symbol("EOS", 4));
+    }
+});
+```
+
+### Multi-index: Delete from Table
+
+This is how to delete an entity from the `balance_table`:
+
+```cpp
+balance_table balances(get_self(), get_self().value);
+
+// check if the user already exists
+auto itr = balances.find(nm.value);
+if ( itr != balances.end() ) {
+    balances.erase(itr);
+}
+```
+
+### Extend Hello Smart Contract with Multi-index
+
+Now, that you know the basic operations you can perform on a multi-index table, you can extend the `hello` contract `hi` action to:
+
+- Keep a list of all accounts that executed at least once the action.
+- Allocate 1 EOS to each user for the first time when the user executes the action.
+- Decrease the asset amount by 0.0001 EOS with every `hi` action execution following the first time.
+- Delete the table row associated with the account sender when `goodbye` action is sent.
+
+The hello.hpp file should look like this:
+
+```cpp
+#include <eosio/eosio.hpp>
+#include <eosio/asset.hpp>
+
+using namespace eosio;
+
+CONTRACT hello : public contract {
+   public:
+      using contract::contract;
+
+      ACTION hi(name nm);
+      ACTION gettime();
+      ACTION goodbye(name nm);
```

Comment:
Please use names that aren't arbitrary for this stuff. 

```
ACTION updaterow
ACTION deleterow
ACTION createrow
```

Will help the dev learn and be able to quickly look at the code they are writing and understand what it does

Reply:
fixed